service/ui_plugin_build_baron.go

```diff
@@ -49,6 +53,44 @@ func bbGetConfig(settings *evergreen.Settings) map[string]evergreen.BuildBaronPr
 	return projects
 }
 
+func bbGetTask(taskId string, execution string) (*task.Task, error) {
+	oldId := fmt.Sprintf("%v_%v", taskId, execution)
+	t, err := task.FindOneOld(task.ById(oldId))
+	if err != nil {
+		return t, err
+	}
+	// if the archived task was not found, we must be looking for the most recent exec
+	if t == nil {
+		t, err = task.FindOne(task.ById(taskId))
```

Comment:
`FindOne` can return `nil, nil`, so you may wish to check for `t == nil`.

Reply:
done